### ISSUE TITLE
Global Objects

### ISSUE THREAD

**Issue by deadcoder0904** [vitalets/react-native-extended-stylesheet#62]
So I am running into an use case where I have default themes.

Suppose I'm creating a `H1` component with default styling & the app also has variety of themes of `H1` component.

Now if I make a global variable `$H1` 

```js
EStyleSheet.build({
     $H1: {
        fontSize: 24,
        fontWeight: 'bold',
        color: 'black'
    }
})
```

Then if I want to use it in styling of `H1` I have to do

```js
{
   H1: {
     fontSize: '$H1.fontSize',
     fontWeight: '$H1.fontWeight',
     color: '$H1.color',
   }
}
```

So this approach is tedious because now when I put some default style in the global object, I've to add it to the local object as well.

So if I can use Global Objects (denoted by `$$`), it would be much easier to make Global Default Styles like

```js
EStyleSheet.build({
   $$H1: {
        fontSize: 24,
        fontWeight: 'bold',
        color: 'black'
   }
})
```

Then if I want to use it in styling of H1 I will do

```js
{
   H1: '$$H1'
}
```

**Comment by vitalets** [comment]
How about using `EStyleSheet.value` to put default style:
```js
EStyleSheet.create({
   H1: EStyleSheet.value('$H1')
});
```
and to extend default style with custom property:
```js
EStyleSheet.create({
   H1: Object.assign({fontSize: 25}, EStyleSheet.value('$H1'))
});
```

**Comment by deadcoder0904** [comment]
Well, I don't realise there exists `value()` 

My bad 😂 but lemme check

**Comment by deadcoder0904** [comment]
Hey also, another problem is I am using themes so I am exporting just plain JS objects. So how would I do this with themeing 🤔

**Comment by vitalets** [comment]
oh, I think I was a bit wrong. It should be: 
```js
EStyleSheet.create({
   H1: () => EStyleSheet.value('$H1')
});
```
and 
```js
EStyleSheet.create({
   H1: () => Object.assign({fontSize: 25}, EStyleSheet.value('$H1'))
});
```

**Comment by deadcoder0904** [comment]
Sorry to make this SO but I thought this feature didn't existed. 

A detailed question would be like this - 

I have 2 themes - `light` & `dark`

### `light.js`

```js
const lightTheme = {
     $H1: {
        fontSize: 24,
        fontWeight: 'bold',
        color: 'black',
        backgroundColor: 'white'
    }
};
```

### `dark.js`

```js
const darkTheme = {
     $H1: {
        fontSize: 24,
        fontWeight: 'bold',
        color: 'white',
        backgroundColor: 'black'
    }
};
```

Then I build my StyleSheet

### `App.js`

```js
EStyleSheet.build(lightTheme)
```

& then use styles in my application like 

```js
{
   H1: {
     fontSize: '$H1.fontSize',
     fontWeight: '$H1.fontWeight',
     color: '$H1.color',
   }
}
```

**Comment by deadcoder0904** [comment]
Man that's what I thought but it wasn't working. Wait I'll check again.

**Comment by deadcoder0904** [comment]
@vitalets I'm getting confused.

I didn't get this one, is `$H1` created in `EStyleSheet.build()` already

```js
EStyleSheet.create({
   H1: () => EStyleSheet.value('$H1')
});
```

The later part I got. This is used while styling a particular component.

```js
EStyleSheet.create({
   H1: () => Object.assign({fontSize: 25}, EStyleSheet.value('$H1'))
});
```

My problem is I am using Themes & it gives error that `Unresolved Variable $H1` bcz it is not created yet. Can u give me an answer for my use case ❔ Is it possible to use value with Themes ❓ 
Bcz Value cannot be used in Exported JS objects. As building takes place later on, before creating.

**Comment by vitalets** [comment]
Yes, `$H1` is assumed to be global variable created by `EStyleSheet.build()`:
```js
EStyleSheet.build({
  $H1: {
    fontSize: 24,
    ...
  }
})
```
Then it can be used in sheet creation:
```js
EStyleSheet.create({
   H1: () => EStyleSheet.value('$H1')
});
```

> Is it possible to use value with Themes

Yes, it is possible.

> Bcz Value cannot be used in Exported JS objects

Yes. `EStyleSheet.value()` is intended to be used inside value as a function, because build occurs later (as you correctly mentioned)

**Comment by vitalets** [comment]
Hmm.. I should re-check it.

**Comment by vitalets** [comment]
@deadcoder0904 I've re-checked - yes, currently *style* as a function is not supported. Only *value* as a function. That's why this does not work:
```js
EStyleSheet.create({
   H1: () => EStyleSheet.value('$H1')
});
```
I've done this locally - it works normally on your code. I will add some tests and release within 1-2 days. Thanks for the ideas =)

**Comment by deadcoder0904** [comment]
So it doesn't work, right ❔ Are u adding this feature ❓

**Comment by vitalets** [comment]
> So it doesn't work, right ❔ Are u adding this feature ❓

Yes. It does not work now. I will release it soon.

**Comment by vitalets** [comment]
Released as 0.8.0. I will appreciate your feedback!

**Comment by deadcoder0904** [comment]
Sorry @vitalets I didn't do it yet. I'll test it now 😄

**Comment by deadcoder0904** [comment]
@vitalets Dude, you are absolutely amazing 👍 
Thank you soooooo much ❤️

**Comment by vitalets** [comment]
You are welcome! 😉
